tests/python_functional/CMakeLists.txt

```diff
@@ -3,3 +3,6 @@ add_custom_target(pytest-self-check
 
 add_custom_target(pytest-check
    COMMAND ${PYTHON_EXECUTABLE} -m pytest ${PROJECT_SOURCE_DIR}/tests/python_functional/functional_tests --installdir=${CMAKE_INSTALL_PREFIX})
+
+add_custom_target(pytest-linters
+   COMMAND find ${PROJECT_SOURCE_DIR}/tests/python_functional/ -name "*.py" | xargs pre-commit run --show-diff-on-failure --config=${PROJECT_SOURCE_DIR}/tests/python_functional/.pre-commit-config.yaml --files)
```

Comment:
For some reason, pre-commit expects to be inside a git repository. If I have the build directory outside of the repository, it fails: `fatal: Not a git repository`. Is it possible to workaround this?

Reply:
As I know pre-commit should use from the git repo itself. 
So it should be an local workaround.

Reply:
I found a fix. Could you add `WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}` to the end of the `add_custom_command` call? I tried and it works for me.

```
add_custom_target(pytest-linters
   COMMAND find ${PROJECT_SOURCE_DIR}/tests/python_functional/ -name "*.py" -not -path "*reports*" -not -path "*tox*" -print0 | xargs -0 pre-commit run --show-diff-on-failure --config=${PROJECT_SOURCE_DIR}/tests/python_functional/.pre-commit-config.yaml --files WORKING_DIRECTORY ${PROJECT_SOURCE_DIR})

```

Reply:
ok, I will check

Reply:
ok, done. Fixed